brut.apktool/apktool-lib/src/main/java/brut/androlib/res/decoder/ResFileDecoder.java

```diff
@@ -119,6 +119,14 @@ public void decode(ResResource res, Directory inDir, Directory outDir)
 
             decode(inDir, inFileName, outDir, outFileName, "xml");
         } catch (AndrolibException ex) {
+            // If we got an error to decode XML, lets assume the file is in raw format.
+            // This is a large assumption, that might increase runtime, but will save us for situations where
+            // XSD files are AXML`d on aapt1, but left in plaintext in aapt2.
+            if (ex.getMessage().equalsIgnoreCase("Could not decode XML")) {
```

Comment:
@iBotPeaches Wouldn't it be better to raise an exception subclass (of `AndroidLibException`) for this issue and check for it instead of checking exception message?

Reply:
Probably a good call, but not sure if a customized exception would be as specific as I want. My perfect world be something like

  - `RawXmlEncounteredException`
  - `AXmlDecodingException`

but unsure if I have the detection for that yet.